domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -574,18 +413,49 @@ public void AddClassButtonPressed()
         }
     }
 
+    public void GetFeedbackButtonPressed()
+    {
+        Debug.Log("Getting feedback");
+        WebCore.GetFeedback();
+    }
+
     /// <summary>
     /// Perform the debug action specified in the body upon the Debug button getting clicked.
     /// </summary>
     public void DebugAction()
     {
         Debug.Log("Debug button clicked!");
         //LoadData();
-        GetCompartmentedRectangles()[0].GetComponent<CompartmentedRectangle>().GetHeader().GetComponent<InputField>()
-          .text = "Rabbit";
+        // GetCompartmentedRectangles()[0].GetComponent<CompartmentedRectangle>().GetHeader().GetComponent<InputField>()
+        //   .text = "Rabbit";
+
+        /*var*/ student = Student.CreateRandom();
+        Debug.Log("student: " + student);
+        Debug.Log("student.Login(): " + student.Login());
+        Debug.Log("student.LoggedIn: " + student.LoggedIn);
+        if (!student.LoggedIn)
+        {
+            infoBox.Warn("Login failed! See console for exact error. (Hint: Double check that WebCORE is running.)");
+            return;
+        }
+        // Logout logic is currently faulty, so uncommenting the lines below will cause the CreateCdm() call to fail
+        // Debug.Log("student.Logout(): " + student.Logout());
+        // Debug.Log("student.LoggedIn: " + student.LoggedIn);
+        // Debug.Log("student.Login() again: " + student.Login());
+        // Debug.Log("student.LoggedIn: " + student.LoggedIn);
+        var cdmCreated = WebCore.CreateCdm(cdmName);
+        Debug.Log("WebCore.CreateCdm(): " + cdmCreated);
+        if (cdmCreated)
+        {
+            infoBox.Info("Class diagram created! You can now add class diagram elements with the diagram editor.");
+        }
+        else
+        {
+            infoBox.Warn("Class diagram creation failed! See console for exact error.");
+        }
     }
 
-    public Dictionary<string, string> getAttrTypeIdsToTypes()
+    public Dictionary<string, string> GetAttrTypeIdsToTypes()
     {
         return attrTypeIdsToTypes;
     }
```

Comment:
We have saved 130 lines of code in this file, while increasing functionality! :moneybag: